scripts/detect_duplicate_features.py

```diff
@@ -0,0 +1,84 @@
+import os
+import argparse
+
+import capa.rules
+import capa.engine as ceng
+
+
+def get_child_features(feature):
+    children = []
+
+    if isinstance(feature, (ceng.And, ceng.Or, ceng.Some)):
+        for child in feature.children:
+            children.extend(get_child_features(child))
+    elif isinstance(feature, (ceng.Subscope, ceng.Range, ceng.Not)):
+        children.extend(get_child_features(feature.child))
+    else:
+        children.append(feature)
+    return children
+
+
+def get_features(rule_path, errors):
+    with open(rule_path, "r") as f:
+        feature_list = []
+        try:
+            new_rule = capa.rules.Rule.from_yaml(f.read())
+            feature_list = get_child_features(new_rule.statement)
+        except Exception as e:
```

Comment:
let's raise the explicit exceptions here and catch them where called, this way we can just return one explicit type

Reply:
and we don't need the errors arguments

Reply:
for files that are not rules, those files are caught as exceptions. I ran the script on one of the rule. I have attached a screenshot of the results.
![Screenshot 2023-05-08 142558](https://user-images.githubusercontent.com/81844215/236781903-bf6519d9-6e15-4c53-876c-4a5c41bab163.png)

Reply:
I used errors argument to return a list of files due to which exceptions occurred.

Reply:
Ok, we can exclude the `.github` directory and also catch the exceptions and print that error directly.

Reply:
> Ok, we can exclude the `.github` directory and also catch the exceptions and print that error directly.

using `get_rules` made it easier. It excludes `.github` by itself and explicitly raises exception for any possible errors.